fix: controller crash when provenance is set only on the global repo

savitaashture opened the pull request:
This commit fixes the controller crash by adding
a nil pointer check before accessing the elements of an object. Issue: https://issues.redhat.com/browse/SRVKP-5571

Verification:

Created Global Repo like below

```
apiVersion: pipelinesascode.tekton.dev/v1alpha1
kind: Repository
metadata:  
  name: pipelines-as-code  
  namespace: openshift-pipelines
spec:  
  settings:    
    pipelinerun_provenance: "default_branch"
```

created local Repo like below

```
apiVersion: pipelinesascode.tekton.dev/v1alpha1
kind: Repository
metadata:  
  name: test  
  namespace: test
spec:  
  settings: {}
  url: https://github.com/savitaashture/test-pac
```

Signed-off-by: Savita Ashture <sashture@redhat.com>

# Changes <!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

<!-- Describe your changes here- ideally you can get that description straight from
your descriptive commit message(s)! -->

# Submitter Checklist

- [ ] 📝 Please ensure your commit message is clear and informative. For guidance on crafting effective commit messages, refer to the How to write a git commit message guide. We prefer the commit message to be included in the PR body itself rather than a link to an external website (ie: Jira ticket).

- [ ] ♽ Before submitting a PR, run make test lint to avoid unnecessary CI processing. For an even more efficient workflow, consider installing [pre-commit](https://pre-commit.com/) and running pre-commit install in the root of this repository.

- [ ] ✨ We use linters to maintain clean and consistent code. Please ensure you've run make lint before submitting a PR. Some linters offer a --fix mode, which can be executed with the command make fix-linters (ensure [markdownlint](https://github.com/DavidAnson/markdownlint) and [golangci-lint](https://github.com/golangci/golangci-lint) tools are installed first).

- [ ] 📖 If you're introducing a user-facing feature or changing existing behavior, please ensure it's properly documented.

- [ ] 🧪 While 100% coverage isn't a requirement, we encourage unit tests for any code changes where possible.

- [ ] 🎁 If feasible, please check if an end-to-end test can be added. See [README](https://github.com/openshift-pipelines/pipelines-as-code/blob/main/test/README.md) for more details.

- [ ] 🔎 If there's any flakiness in the CI tests, don't necessarily ignore it. It's better to address the issue before merging, or provide a valid reason to bypass it if fixing isn't possible (e.g., token rate limitations).

savitaashture committed:
fix: controller crash when provenance is set only on the global repo

This commit fixes the controller crash by adding
a nil pointer check before accessing the elements of an object.
Issue: https://issues.redhat.com/browse/SRVKP-5571

Signed-off-by: Savita Ashture <sashture@redhat.com>

enarha commented on test/gitea_test.go:
```diff
@@ -637,6 +637,76 @@ func TestGiteaProvenance(t *testing.T) {
+
+	_, f := tgitea.TestPR(t, topts)
+	defer f()
+	targetRef := topts.TargetRefName
```
What will be the value of targetRef here?

savitaashture replied:
Its the source branch value from where PR is sent

enarha replied:
OK. I was just trying to understand if it is different from the default branch and it sounds like it is (please correct me if I'm wrong). Thanks.

savitaashture replied:
actually current test doesn't check the actualProvenance feature so updating tests

enarha replied:
I thought it does, that's what I was trying to figure, but now thinking again about it, no, it does not.

enarha commented:
/lgtm

savitaashture commented:
I have corrected e2e 

But why linter is failing i need to check

piyush-garg committed:
Fix vale config file according to latest version

This will fix vale config file to compatible with latest version as
it was throwing error that IgnoredScopes and SkippedScopes to be
defined before syntx options

savitaashture commented:
/test linters

savitaashture commented:
@zakisk @enarha CI is green 
PTAL Thank you

enarha commented:
I like it better with the updated test. Thanks.

savitaashture commented:
/test go-testing
